File: Flutter/routine_flutter/lib/ui/edit/period.dart
enum Period { DAY, WEEK, MONTH }

Period findPeriod(String value) =>
    Period.values.firstWhere((element) => element.name == value);

extension PeriodExt on Period {
  static const periodNameMap = {
    Period.DAY: 'day',
    Period.WEEK: 'week',
    Period.MONTH: 'month'
  };

  int get id => this.index;

  String get name => periodNameMap[this];
}

File: iOS/UIKit/Routine/Models/Period.swift
//
//  Period.swift
//  Routine
//
//  Created by Vadzim Karonchyk on 3/26/20.
//  Copyright © 2020 Instinctools. All rights reserved.
//

import Foundation

enum Period: String, CaseIterable {
    case day = "DAY"
    case week = "WEEK"
    case month = "MONTH"
    case year = "YEAR"
        
    var calendarComponent: Calendar.Component {
        switch self {
        case .day:
            return .day
        case .week:
            return .weekOfMonth
        case .month:
            return .month
        case .year:
            return .year
        }
    }
}

extension Optional where Wrapped == Period {
    var orDefault: Period {
        return self ?? .day
    }
}
